hyppo/independence/hhg.py

```diff
@@ -224,3 +309,97 @@ def _pearson_stat(distx, disty):  # pragma: no cover
                     S[i, j] = ((n - 2) * (t12 * t21 - t11 * t22) ** 2) / denom
 
     return S
+
+
+def _point_distance(self, x, y, zx, zy, **kwargs):
+    """
+    For fast HHG,
+    Returns a collection of distances between sample points and chosen centre point
+    """
+    distx = cdist(zx, x, metric=self.compute_distance, **kwargs)
+    disty = cdist(zy, y, metric=self.compute_distance, **kwargs)
+
+    return distx, disty
+
+
+def hoeffdingsD(x, y):
+    """
+    For fast HHG, calculates the Hoeffding's dependence statistic
+    """
+    xin = x
+    yin = y
+    # crop data to the smallest array, length have to be equal
+    if len(xin) < len(yin):
+        yin = yin[: len(xin)]
+    if len(xin) > len(yin):
+        xin = xin[: len(yin)]
+    # dropna
+    x = xin[~(np.isnan(xin) | np.isnan(yin))]
+    y = yin[~(np.isnan(xin) | np.isnan(yin))]
+
+    # undersampling if length too long
+    lenx = len(x)
+    if lenx > 99999:
+        factor = math.ceil(lenx / 100000)
+        x = x[::factor]
+        y = y[::factor]
+
+    R = rankdata(x)
+    S = rankdata(y)
+
+    # core processing
+    N = x.shape
+    dico = {(np.nan, np.nan): np.nan}
+    dicoRin = {np.nan: np.nan}
+    dicoSin = {np.nan: np.nan}
+    dicoRless = {np.nan: np.nan}
+    dicoSless = {np.nan: np.nan}
+    Q = np.ones(N[0])
+
+    i = 0
+    for r, s in np.nditer([R, S]):
+        r = float(r)
```

Comment:
unnecessary, `x` and `y` are full of floats

Reply:
Yep, removed.